hiera.yaml

```
@@ -0,0 +1,16 @@
+---
+version: 5
+
+defaults:
+  datadir: 'data'
+  data_hash: 'yaml_data'
+
+hierarchy:
+  - name: 'Operating System Major Release'
+    path: '%{facts.os.family}/%{facts.os.release.major}.yaml'
```

I'd go for:
```suggestion
    path: 'os/%{facts.os.name}/%{facts.os.release.major}.yaml'
```

I'd also add a hierachy that looks at `os/%{facts.os.name}.yaml` (or similar).

---

updated